lib/abbreviato/truncated_sax_document.rb

```diff
@@ -0,0 +1,194 @@
+require 'nokogiri'
+require 'htmlentities'
+
+class TruncatedSaxDocument < Nokogiri::XML::SAX::Document
+  IGNORABLE_TAGS = %w[html head body].freeze
```

Comment:
why are these ignorable? do they get generated by Nokogiri elsewhere? do we account for their bytesizes?